StylePropertyMapReadOnly get method implemented, parse method implemented for StyleValue and NumberValue

njlawton opened the pull request.

Commit:
Merge branch 'master' into Style-map-get

Commit:
Created the base of the get method in the StylePropertyMapReadOnly class

Commit:
Implemented parse method in StyleValue class
This method while implemented is currently not functional as no StyleValue subclasses currently implement parse

Commit:
Implemented getValidStyleValuesArray method in PropertyDictionary
This method returns an array of StyleValue constructors supported by a CSS property

Commit:
Implemented parse for the NumberValue class
this also allowed us to test the get method for StylePropertyMapReadOnly, small changes were also made to this method to optimise it
The isValidKeyword method in the PropertyDictionary class was also made public as it is usefull for methods outside its class.

Commit:
Added new test cases to ensure NumberValues parse works correctly
Also added a regex checker to the parse method

Commit:
Merge branch 'master' into Style-map-get

Commit:
Changed regex expresion used in the parse method of number value
this new regex expresion will not allow numbers containing decimal points after the exponential e in numbers

wilddamon commented on test/js/computed-style-property-map.js:
```diff
@@ -8,20 +8,18 @@ suite('Computed StylePropertyMap', function() {
-    assert.equal(opacity.cssString, '0.5');
+    var propertyStyleValue = computedStyleMap.get('lemon');
+
+    assert.equal(propertyStyleValue, null);
```
Let's throw an exception here instead of just returning null.

wilddamon commented on src/style-value.js:
```diff
@@ -16,15 +16,24 @@
   function StyleValue() {}
 
-  // TODO: Include parsing logic here.
   StyleValue.parse = function(property, value) {
```
How will this work if value should become a sequence? Don't need to address this now, just need add a TODO.

wilddamon commented on test/js/inline-style-property-map.js:
```diff
@@ -21,6 +21,16 @@ suite('Inline StylePropertyMap', function() {
     assert.strictEqual(this.element.style['height'], simpleLength.cssString);
   });
 
+  test('The set method successfully sets the CSS string of the list of StyleValues on property ' +
```
This seems unrelated? If you have changes that you would like to make to unrelated parts of the codebase, it's good practice to put them in separate pull requests.

wilddamon commented on test/js/number-value.js:
```diff
@@ -20,4 +20,26 @@ suite('NumberValue', function() {
+    assert.strictEqual(NumberValue.parse('3px'), null);
+    assert.strictEqual(NumberValue.parse('scale(3, -1)'), null);
+    assert.strictEqual(NumberValue.parse('1e-.0'), null);
+    assert.strictEqual(NumberValue.parse('1e4.0'), null);
```
What about `'6 someOtherRandomString'`?

wilddamon commented on test/js/property-dictionary.js:
```diff
@@ -65,4 +65,15 @@ suite('PropertyDictionary', function() {
     assert.throw(function () {cssPropertyDictionary.getListValueSeparator('height')}, TypeError);
   });
+
+  test('The getValidStyleValuesArray should return an array of constructors for all StyleValue types accepted by that propertty', function() {
```
Can leave off "The" at the start.
s/propertty/property (extra t)

'getValidStyleValuesArray should return an array of constructors for all StyleValue types accepted by that property'

wilddamon commented on src/number-value.js:
```diff
@@ -23,7 +23,13 @@
   internal.inherit(NumberValue, internal.StyleValue);
 
+  NumberValue.parse = function(value) {
+    if (/^\s*[-+]?(\d*\.)?\d+(e[-+]?\d+)?\s*$/.test(value)) {
```
Lets move this regex out into a parser file like in my pull request. There will be lots of other properties that will need to parse numbers as part of their parsing, and it makes sense to share this regex with them.

wilddamon commented on src/style-value.js:
```diff
@@ -16,15 +16,24 @@
+    }
+
+    if (propertyDictionary().isValidKeyword(property, value)) {
+      return Keyword.parse(value);
```
Isn't this just the same as `new KeywordValue(value)`? It's clearer just to use the constructor in this case.

wilddamon commented on src/style-property-map-readonly.js:
```diff
@@ -19,77 +19,16 @@
+      return null;
     }
 
+    return StyleValue.parse(property, propertyString);
```
As discussed, let's add the handling the possibility of arrays coming out of parse now, so

``` javascript
  var parsed = StyleValue.parse(property, propertyString);
  if (typeof parsed == 'Array') {
    return parsed[0];
  }
  return parsed;
```

Commit:
Made changes suggested in the pull request
Major change made to the get method that now supports sequnces of list values however since sequences have not been implemented in StyleValue.parse this will currently never happen

njlawton commented:
Not finished yet just needed to clear changes already made

Commit:
Added parser class so that parsing logic can be reused
added isNumberValueString method and used this method in the parse method of the NumberValue

Commit:
Moved test case irrelevant to this branch

Commit:
Merge branch 'master' into Style-map-get

Commit:
Changed variable isNumberValueString in parsing.js to private

Commit:
Fixed Small code formmat error in src/parsing.js

wilddamon commented on src/style-property-map-readonly.js:
```diff
@@ -19,77 +19,20 @@
-    var value = this._styleObject[property];
-    if (!value) {
+    propertyString = this._styleObject[property];
+    if (propertyString == '') {
```
You can still write (and I think it's clearer since `'' == false == null` in javascript:

``` javascript
if (!propertyString)
```

Commit:
Changed if statement wording to improve understandability in the StylePropertyMapReadOnly class
